types/jasmine/index.d.ts

```diff
@@ -316,15 +316,12 @@ declare namespace jasmine {
     function formatErrorMsg(domain: string, usage: string): (msg: string) => string;
 
     interface Any extends AsymmetricMatcher<any> {
-        (...params: any[]): any; // jasmine.Any can also be a function
         new (expectedClass: any): any;
-
-        jasmineMatches(other: any): boolean;
         jasmineToString(): string;
     }
 
     interface AsymmetricMatcher<TValue> {
-      asymmetricMatch(other: TValue, customTesters: ReadonlyArray<CustomEqualityTester>): boolean;
+      asymmetricMatch(other: TValue, matchersUtil?: MatchersUtil): boolean;
```

Comment:
The old form of `asymmetricMatch` still works for now, although it will cause a deprecation warning in 3.99 and break in 4.0. (Basically, for now Jasmine passes an object that works as both a `MatchersUtil` and an array of custom equality testes). I know DefinitelyTyped definitions don't follow semver and I don't have a strong opinion about how to handle this change, but I wanted to point it out.

Reply:
I went through this mess when I was upgrading Jasmine 3.5 to 3.6 in our organization. I would go with the approach of putting the more correct type of MatchersUtil, but keeping the old function around and marking it with `@deprecated`. Then when the release of Jasmine that kills custom equality tester comes around, you can just remove the deprecated function.